nav2_docking/opennav_docking/src/simple_charging_dock.cpp

```diff
@@ -322,6 +350,88 @@ void SimpleChargingDock::jointStateCallback(const sensor_msgs::msg::JointState::
   is_stalled_ = (velocity < stall_velocity_threshold_) && (effort > stall_effort_threshold_);
 }
 
+void SimpleChargingDock::startDetection()
+{
+  // Skip if already starting or ON
+  if (detector_state_ != DetectorState::OFF) {return;}
```

Comment:
```suggestion
  if (detector_state_ == DetectorState::ON) {return;}
```

This seems more rational (though I think you're going to change this to a bool anyway)